### PR TITLE
Make Quarkus more configurable

### PR DESCRIPTION
**Pull request by martin-fleck-at**
- Upgrade to Quarkus 2.12 for additional properties
- Increase default http read timeout to 5M
- Allow customization of service port (default: 8081)
- Print Quarkus configuration to debug on startup

### CODE REVIEW

File: java/service/org.eclipse.theia.cloud.service/src/main/java/org/eclipse/theia/cloud/service/ApplicationLifecycleListener.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+import io.quarkus.runtime.StartupEvent;
+import io.vertx.core.json.JsonObject;
+
+@ApplicationScoped
+public class ApplicationLifecycleListener {
+    protected Logger logger;
+
+    protected void onStart(@Observes StartupEvent event) {
+	logger = Logger.getLogger(getClass().getSuperclass());
```
Comment (jfaltermeier) [inline java/service/org.eclipse.theia.cloud.service/src/main/java/org/eclipse/theia/cloud/service/ApplicationLifecycleListener.java:40]: The superclass is object, so logs will look like this: `2022-09-05 09:15:37,368 DEBUG [jav.lan.Object] (main) Resulting Quarkus Configuration` which is strange

File: java/service/org.eclipse.theia.cloud.service/src/main/java/org/eclipse/theia/cloud/service/ApplicationLifecycleListener.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    private void logConfiguration() {
+	logConfigurationSources();
+	logQuarkusConfiguration();
+    }
+
+    private void logConfigurationSources() {
+	for (ConfigSource configSource : ConfigProvider.getConfig().getConfigSources()) {
+	    logger.debug(configSource.getName() + " (" + configSource.getOrdinal() + ")");
```
Comment (jfaltermeier) [inline java/service/org.eclipse.theia.cloud.service/src/main/java/org/eclipse/theia/cloud/service/ApplicationLifecycleListener.java:51]: I think we could use info logging, when its only printed on startup

File: java/service/org.eclipse.theia.cloud.service/pom.xml [hunk 3]
```diff
@@ -13,6 +13,15 @@
 		<relativePath>../../common/maven-conf/</relativePath>
 	</parent>
 
+	<properties>
+		<skipITs>true</skipITs>
+		<quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
+		<quarkus.platform.group-id>io.quarkus.platform</quarkus.platform.group-id>
+		<!-- For some reason quarkus.platform.version must be defined here and
```
Comment (jfaltermeier) [inline java/service/org.eclipse.theia.cloud.service/pom.xml:20]: Could you please elaborate on this. 
I moved the three properties back up to `common/maven-conf`and launching locally and building the docker images still worked. So I would rather like to keep the properties up in the global maven-conf file.

**Review by jfaltermeier — COMMENTED** [review]
Thank you. Changes work as expected as far as I could see. 
I have some comments though.

File: java/service/org.eclipse.theia.cloud.service/src/main/java/org/eclipse/theia/cloud/service/ApplicationLifecycleListener.java (hunk 1 above)
Comment (martin-fleck-at) [reply java/service/org.eclipse.theia.cloud.service/src/main/java/org/eclipse/theia/cloud/service/ApplicationLifecycleListener.java:40]: Ah, good catch, sorry for that!

File: java/service/org.eclipse.theia.cloud.service/src/main/java/org/eclipse/theia/cloud/service/ApplicationLifecycleListener.java (hunk 2 above)
Comment (martin-fleck-at) [reply java/service/org.eclipse.theia.cloud.service/src/main/java/org/eclipse/theia/cloud/service/ApplicationLifecycleListener.java:51]: Fine by me, I changed it to `info`.

File: java/service/org.eclipse.theia.cloud.service/pom.xml (hunk 3 above)
Comment (martin-fleck-at) [reply java/service/org.eclipse.theia.cloud.service/pom.xml:20]: Hmm... seems the issue I was facing is only locally. I always received the following error:
```
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------< org.eclipse.theia.cloud:service >-------------------
[INFO] Building service 0.7.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ service ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 2 resources
[INFO] 
[INFO] --- quarkus-maven-plugin:2.12.0.Final:generate-code (default) @ service ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.040 s
[INFO] Finished at: 2022-09-05T11:44:17+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.12.0.Final:generate-code (default) on project service: Quarkus code generation phase has failed: io.quarkus.deployment.CodeGenerator.initAndRun(java.lang.ClassLoader, io.quarkus.paths.PathCollection, java.nio.file.Path, java.nio.file.Path, java.util.function.Consumer, io.quarkus.bootstrap.model.ApplicationModel, java.util.Properties, java.lang.String, boolean) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

Maybe it is a difference in maven version (I use 3.8.5 and the image uses 3.8.4). In any case, if we build through Docker this should not be an issue so I reverted the change and moved them back.

**Pull request merged**